pkg/controllerutils/equality_test.go

```diff
@@ -81,6 +81,32 @@ var _ = Describe("ObjectsEqual", func() {
 		equal := ObjectsEqual(obj1, obj2)
 		Expect(equal).To(BeFalse())
 	})
+
+	It("asserts equality on two proto.Message objects even if they are passed in by value", func() {
+		obj1 := &things_test_io_v1.Paint{
+			Spec: things_test_io_v1.PaintSpec{
+				Color: &things_test_io_v1.PaintColor{
+					Hue: "red",
+				},
+				PaintType: &things_test_io_v1.PaintSpec_Oil{
+					Oil: nil,
+				},
+			},
+		}
+		obj1.Spec.ProtoReflect().SetUnknown([]byte(""))
```

Comment:
Previously, when the Spec was being passed in by value, then line
```
protoVal1, isProto := val1.(proto.Message)
```
would return isProto as false, and use reflect.DeepEqual instead of the correct proto.Equal, which allows us to treat empty .status.atomicInfo fields as the same as nil .status.atomicInfo fields.